### ISSUE TITLE
Fix inconsistent and unusual naming conventions of CSS attributes

### ISSUE THREAD

**Issue by Ishirak** [versionpress/versionpress#1084]
This issue is part of a frontend code review, see #1083.

The front end codebase contains inconsistent and unusual names for the class and id CSS attributes.

For example, the BulkActionPanel component contains following classNames:
- BulkActionPanel
- bulkactions
- BulkActionPanel-note hide
- screen-reader-text

and following id:
- BulkActionPanel-selector-top

Inconsistency in naming conventions makes these CSS attributes hard to follow. Author probably intend to specify component's name in the className or Id attributes, but overall it lacks consistency.

So my proposal is to make names consistent, for example:
- BulkActionPanel → bulk-action-panel
- bulkactions → bulk-actions
- BulkActionPanel-note hide → bulk-action-panel-note hide
- screen-reader-text → _it is fine_

**Comment by vasek17** [comment]
My idea about naming CSS rules is based on Block-Element-Modifier. React component and Block in BEM have roughly the same meaning so I use the component's name as first part of the CSS name.
I would prefer not to have more than one name for one component (e.g. `BulkActionPanel` and `bulk-action-panel`). In my opinion it is ok to use directly the component name.

There are some inconsitencies due to compatibility with WordPress styles (e.g. `bulkactions`, `screen-reader-text`). We may change the overal design and therefore we would drop this compatibility styles in the future.

**Comment by borekb** [comment]
I can see a point on both sides. Maybe we can draw inspiration from the Calypso project? How do they deal with stylesheets?

**Comment by vasek17** [comment]
Calypso uses component's display name in `PascalCase` and the class names in "dash-case".

They use something like [SUIT CSS](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md) for organizing stylesheets.
It would look like this:

`component-name/component-name.jsx`
```JavaScript
class ComponentName extends React.Component {
    ...
}
```

`component-name/style.scss`
```CSS
.component-name {
    ...
}
.component-name__element.has-smth {
    ...
}
.component-name.is-hidden {
    ...
}
```

**Comment by vasek17** [comment]
Current VersionPress way would look like this:

`ComponentName/ComponentName.react.tsx`
```JavaScript
class ComponentName extends React.Component {
    ...
}
```

`ComponentName/ComponentName.less`
```CSS
.ComponentName {
    ...
}
.ComponentName-element.ComponentName--hasSmth {
    ...
}
.ComponentName.ComponentName--hidden {
    ...
}
```

**Comment by borekb** [comment]
Great comparison, thank you. In Calypso's approach, I don't like that there are hundreds of `style.scss` files in the project – I'd be interested to see how they "navigate to file" in their editors / IDEs. It seems much better to have a component name in the stylesheet.

Other than that, I don't really have a strong preference in either the `dash-style` or `PascalCase` style. It's true that in most "simple" web project, the dash-convention is more common but here we're dealing with components and at least it's immediately obvious which class name is for a component and which is potentially "inherited" from WordPress styles (e.g., `BulkActionPanel` vs. `bulkactions`).

**Comment by Ishirak** [comment]
Yes, we're dealing with components, but these components React.js transforms to the "normal" DOM elements. 

My only complaint is that `PascalCase` style in the CSS attributes is highly irregular and I don't know any website who uses this style. I'm sure that there are few, so please share them, if you know about any of them.

If you are concerned about WordPress inheritance, there is always an option to prefix name with `vp-`, but I don't like this option at all.

**Comment by vasek17** [comment]
Ad WP inheritance - The reason is to have some elements styled the standard WordPress way so the classnames should stay the same.
EDIT: I thought you were writing about `wp-` not `vp-`. However I don't think we should use any prefix for all styles.

The best approach would be something like [CSS modules](https://github.com/css-modules/css-modules) but i don't know if they are ready to use in production yet. [This](https://github.com/gajus/react-css-modules) looks promising but I don't have any experience with it.

I don't see a problem in PascalCase names, it is irregular but it makes perfect sense in React application. 
The dash-case has a few problems for me:
1. Double-click on an identifier should select the whole identifier not only a part
2. Dashes are currently used as a separator of Block-Element-Modifier parts. It would be needed to use something else (and I don't like to use underscores as separators - see 1.)
3. The component name and the name of CSS rule is different. In my opinion it is be better to use the same names in our situation (CSS is created together with code).

**Comment by Ishirak** [comment]
I don't know much about CSS modules, I only know that it exists, but I have never used it. Another way could be [Aphrodite](https://github.com/Khan/aphrodite). I think that either approach would be great because our front end isn't "normal" website, but it is a plugin, which has to "behave" itself among WP and other plugins.

If you insist on using PascalCase then I don't have any _strong_ arguments beside which I've already said in this thread.

**Comment by borekb** [comment]
This really feels like a subjective issue and can be argued both ways. Fortunately, it's not that crucial for the project. You might try to solicit more feedback from @JanVoracek but if the vote is roughly equal / neutral, let's not spend too much time on this.

**Comment by JanVoracek** [comment]
I quite like the PascalCase as we can easily recognize WP-related classes. But both styles are OK.
